roles/openshift_azure/tasks/copy_md_to_vhd.yml

```diff
@@ -0,0 +1,104 @@
+---
+- debug:
+    msg: "{{ openshift_azure_vm_name }}"
+    verbosity: 1
+
+- name: find the disk
+  azure_rm_managed_disk_facts:
+    resource_group: "{{ openshift_azure_resource_group_name }}"
+    name: "{{ item }}"
+  register: diskout
+  with_items:
+  - "{{ openshift_azure_vm_name }}.vhd"
+  - "{{ openshift_azure_vm_name }}-datadisk-0"
+
+- name: set the diskid as a fact
+  set_fact:
+    l_managed_disk_ids: "{{ diskout | json_query('results[*].ansible_facts.azure_managed_disk[0].id') }}"
+
+- debug:
+    msg: "{{ l_managed_disk_ids }}"
+    verbosity: 1
+
+- name: create a SAS url
+  command: "az disk grant-access --ids {{ item }} --duration-in-seconds 3600 --output=json"
+  register: grantout
+  with_items: "{{ l_managed_disk_ids }}"
+
+- debug: var=grantout
+
+
+- name: parse json and set sas_data_list
+  set_fact:
+    l_sas_data_list: "{{ grantout | json_query('results[*].stdout') | openshift_azure_convert_list_to_json }}"
```

Comment:
out of interest how does openshift_azure_convert_list_to_json differ from jinja2 to_json ?

Reply:
It takes a list of results since there are multiple disks and appends them to a single list.